src/views/forms/TriggersForm/LimitPriceInputs.tsx

```diff
@@ -15,26 +21,80 @@ import { FormInput } from '@/components/FormInput';
 import { Tag } from '@/components/Tag';
 import { WithTooltip } from '@/components/WithTooltip';
 
+import { getTriggerOrdersInputs } from '@/state/inputsSelectors';
+
+import abacusStateManager from '@/lib/abacus';
+import { MustBigNumber } from '@/lib/numbers';
+
 type ElementProps = {
+  existsLimitOrder: boolean;
+  multipleTakeProfitOrders: boolean; 
+  multipleStopLossOrders: boolean;
   tickSizeDecimals?: number;
 };
 
 type StyleProps = {
   className?: string;
 };
 
-export const LimitPriceInputs = ({ tickSizeDecimals, className }: ElementProps & StyleProps) => {
+export const LimitPriceInputs = ({
+  existsLimitOrder,
+  multipleTakeProfitOrders,
+  multipleStopLossOrders,
+  tickSizeDecimals,
+  className,
+}: ElementProps & StyleProps) => {
   const stringGetter = useStringGetter();
 
+  const { stopLossOrder, takeProfitOrder } =
+    useSelector(getTriggerOrdersInputs, shallowEqual) || {};
+
   const [shouldShowLimitPrice, setShouldShowLimitPrice] = useState(false);
 
+  const decimals = tickSizeDecimals ?? USD_DECIMALS;
+
+  useEffect(() => {
+    setShouldShowLimitPrice(existsLimitOrder);
+  }, [existsLimitOrder]);
```

Comment:
umm i think this is a little confusing T__T is it because the state is delayed?

Reply:
yeah - it's cuz the state is technically read from abacus, but that's initially set by the front end dialog component mounting (which mounts last because it's the parent component to all the individual children) :( I got a weird flickering thing without this, but def open to better ideas!!

Reply:
i see i see - if this is just the initial state, can we just replace this with
```
const [shouldShowLimitPrice, setShouldShowLimitPrice] = useState(existsLimitOrder);
```

also right now when i toggle limit price, it will disappear if i uncheck and check it, is that expected?

Reply:
oops responded here: https://github.com/dydxprotocol/v4-web/pull/426#discussion_r1556497768